ipaserver/plugins/cert.py

```diff
@@ -771,6 +771,17 @@ def execute(self, csr, all=False, raw=False, chain=False, **kw):
         cn = cns[-1].value  # "most specific" is end of list
 
         if principal_type in (SERVICE, HOST):
+
+            has_dns_in_san_ext = False
+            if ext_san:
+                for gn in x509.process_othernames(ext_san.value):
+                    if isinstance(gn, cryptography.x509.general_name.DNSName):
+                        has_dns_in_san_ext = True
+
+            if not ext_san or not has_dns_in_san_ext:
+                print('Warning: The SAN extension '
+                      'should be provided. Please, check the RFC 2818.')
```

Comment:
I don't think we need to reference the RFC.

Ideally we should suggest to the user what to do, but because we don't know what tool created the CSR
(certmonger, certutil, openssl, ...?) it is a bit hard to give concrete steps to user about how to solve their problem.

I would say something like: "Certificate request for a {host,service} certificate did not contain Subject Alternative Name, which is required for many use cases."

This warning will be suprious if the profile uses the new CommonNameToSANDefault component (assuming the CN looks like a DNS name).   I have a PR to add this component to `caIPAserviceCert` (the default profile) for new installs only.

I'm just not sure if this won't create more support load than it avoids.  It is hard to guess what will happen.